src/corehost/common/pal.unix.cpp

```diff
@@ -204,12 +204,18 @@ bool locate_dotnet_on_path(pal::string_t* dotnet_exe)
         append_path(&tok, _X("dotnet"));
         if (pal::realpath(&tok) && is_executable(tok))
         {
-            *dotnet_exe = tok;
+            dotnet_exes->push_back(tok);
```

Comment:
When are you ever returning more than one here?  You are returning 'true' as soon as you push something on the list.

Do we really need to handle more than one?